### PR TITLE
[GLUTEN-2934][CH] Fix error knownfloatingpointnormalized

### PR DESCRIPTION
**Pull request by loneylee**
## What changes were proposed in this pull request?

(Fixes: \#2934)

## How was this patch tested?

test by ut

### CODE REVIEW

**Comment by zzcclp** [comment]
@PHILO-HE please help to check whether this fix is suitable for Velox backend.

File: backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/utils/CHExecUtil.scala [hunk 1]
```diff
@@ -177,13 +177,14 @@ object CHExecUtil extends Logging {
...
-    val hashFields = partitoining.expressions.map {
-      case a: Attribute =>
-        BindReferences
-          .bindReference(a, childOutput)
-          .asInstanceOf[BoundReference]
-          .ordinal
-    }
+    val hashFields =
+      partitoining.expressions.map(ConverterUtils.getAttrFromExpr(_).toAttribute).map {
+        case a: Attribute =>
```
Comment (PHILO-HE) [inline backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/utils/CHExecUtil.scala:182]: Nit:
It looks unnecessary to keep `case a: Attribute =>`, as the first map's output is always `Attribute` type.

**Review by PHILO-HE — APPROVED** [review]
Just tested velox backend in main branch. Looks no issue found at least for this test case.
@zzcclp, thanks for letting me know this.

File: backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/utils/CHExecUtil.scala (hunk 1 above)
Comment (loneylee) [reply backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/utils/CHExecUtil.scala]: fixed

**Review by PHILO-HE — APPROVED** [review]
Thanks!

**Pull request merged**